### ISSUE TITLE
Wanted 'Tab' Throwing Error

### ISSUE THREAD

**Issue by razambon** [evilhero/mylar#910]
Getting the following error whenever I try to go to the wanted tab.  (the /upcoming page) Not sure what's causing it.

	Error !
	TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
		41 <tr class="gradeZ">
		42 <td id="select"><input type="checkbox" name="${issue['IssueID']}" class="checkbox" /></td>
		43 <td id="comicname"><a href="comicDetails?ComicID=${issue['ComicID']}">
		44 <%
		45 if any(d['IssueID'] == str(issue['IssueID']) for d in ann_list):
		46 adjcomicname = issue['ComicName'] + ' Annual'
		47 else:
		48 adjcomicname = issue['ComicName']
		49 endif
	C:\Program Files\mylar\data/interfaces/default/upcoming.html, line 44:
	<%
	C:\Program Files\mylar\data/interfaces/default/base.html, line 82:
	${next.body()}
	C:\Program Files\mylar\mako\runtime.py, line 718:
	callable_(context, *args, **kwargs)
	C:\Program Files\mylar\mako\runtime.py, line 692:
	_exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
	C:\Program Files\mylar\mako\runtime.py, line 660:
	**_kwargs_for_callable(callable_, data))
	C:\Program Files\mylar\mako\template.py, line 296:
	return runtime._render(self, self.callable_, args, data)
	C:\Program Files\mylar\mylar\webserve.py, line 49:
	return template.render(**kwargs)

**Comment by rupaschomaker** [comment]
I'm also experiencing this one...  I don't recall doing anything like you describe, but I also don't know how long the wanted link hasn't been working for me (not something I normally use).

**Comment by evilhero** [comment]
Can either of you throw your mylar.db file up somewhere and I'll sideload it into mine and try and see what's happening - right now, I can't duplicate this cause I can't reference the same things you guys are.

**Comment by rupaschomaker** [comment]
https://storage.rupa.com:5001/fbsharing/PD3AePfC

Link should be valid for a week.

**Comment by evilhero** [comment]
I'll grab it in the next hr or two and see what I can figure out.. Thanks for linking the db!

**Comment by razambon** [comment]
I actually can't bring up my wanted list at all - so I can't determine what's throwing the error.

**Comment by evilhero** [comment]
Ok, well I found the problem issues:
```
sqlite> select * from annuals where Status='Wanted';
297223||||Wanted||||||||||
297222||||Wanted||||||||||
```
and then referencing the IssueID's in the issues table:
```
sqlite> select * from issues where IssueID='297223';
297223|Red Hood and the Outlaws|shot through the heart, and who's to blame?|2||Archived||42919||2011-10-19||2011-12-27|2000||||
sqlite> select * from issues where IssueID='297222';
297222|Catwoman|i could say that i'll sleep better tomorrow, but that's a lie|2||Downloaded||42722||2011-10-19|Catwoman 02 (2011).cbr|2011-12-26|2000|27644970|||
```

Normally on startup, Mylar clears out any empty entries within the db tables to ensure clean starts and that the data is clean as well. The problem, in this case, was that there were entries in the annuals table that weren't supposed to be there (they were supposed to be in the issues table only). Now why this happened, I really don't know - it could have been some stagnant entries from long ago that just resurfaced openly with some changes that were done recently, or it could have been a misglitch that might have been subsequently fixed.

The end result: I added a clear out for the annuals table on startup, tested it with rupaschomaker's db (thanks for sending that!), and was able to access the Upcoming screen just fine.

Expect a commit for this up shortly.
